packages/components/Notification/src/Notification.tsx

```diff
@@ -51,18 +50,46 @@ export const Notification = compose<NotificationType>({
     }, ['isBar']);
 
     return (final: NotificationProps, ...children: React.ReactNode[]) => {
-      const { variant, icon, title, action, ...rest } = mergeProps(userProps, final);
-      const iconProps = createIconProps(icon);
+      const { variant, icon, title, action, onActionPress, ...rest } = mergeProps(userProps, final);
       const mergedProps = mergeProps<PressableProps>(rest, rootStyle);
 
+      const iconProps = createIconProps(icon);
+      const dismissSvg: React.FunctionComponent<SvgProps> = (props: SvgProps) => {
+        return getDismissSvg(props.color);
+      };
+      const svgProps: SvgIconProps = {
+        src: dismissSvg,
+      };
+      const dismissIconProps = createIconProps({ svgSource: svgProps, width: 20, height: 20 });
+
       return (
         <Slots.root {...mergedProps}>
           {icon && <Slots.icon {...iconProps} />}
           <Slots.contentContainer>
             {title && <Slots.title>{title}</Slots.title>}
             <Slots.message style={messageStyle}>{children}</Slots.message>
           </Slots.contentContainer>
-          {action ? <Slots.action>{action}</Slots.action> : !isBar && <Slots.dismissIcon>{getDismissIconPath()}</Slots.dismissIcon>}
+          {onActionPress ? (
+            action ? (
+              <Slots.action onClick={onActionPress}>{action}</Slots.action>
+            ) : (
+              <Slots.action
+                icon={dismissIconProps}
+                onClick={() => {
+                  console.log('Dismiss button tapped');
+                }}
+              />
+            )
+          ) : (
+            !isBar && (
+              <Slots.action
+                icon={dismissIconProps}
+                onClick={() => {
+                  console.log('Dismiss button tapped');
+                }}
+              />
+            )
+          )}
```

Comment:
In general this section feels it got very complicated. Can we do any simplification here?

Reply:
Yeah it changed quite a bit after I talked to Jeanie and realized I had been reasoning about the action button wrong. At first I had assumed that if there is no action specified, a dismiss button is added (that's how it was). However, she differentiated between the action's text and the callback function itself. So if there's a callback and a text, it should render the text (line 74) but it there's a callback and no text, it should render a dismiss button (line 76-81). And the finally, toasts should have an action so if there's no callback, it should have a dismiss button (line 84-91)

Reply:
Thanks, this looks cleaner now :) 

Could you perhaps leave a comment in code somewhere describing when to show the dismiss vs action button based on the props passed? Should we simplify to some booleans like `const shouldShowDismissIcon = blahblah`?